jax/util.py

```diff
@@ -152,7 +151,10 @@ def __missing__(self, key):
 
 
 def prod(xs):
-  return functools.reduce(mul, xs, 1)
+  out = 1
```

Comment:
I'm not sure if `functools.reduce` is `__builtins__.reduce` (aka `six.moves.reduce`), but I timed this on Python 2 just now and this new implementation (writing out the loop) is indeed faster than using `__builtins__.reduce` on Python 2.

Reply:
It was also noticeably faster on Python 3 on my local machine. It isn't that important but I figured I'd check it in; arguably it's clearer code (but precious vertical space!)